models/filtermodel.cpp

```diff
@@ -35,6 +37,37 @@ void FilterModel::setSourceModel(DocumentListModel *model)
     QSortFilterProxyModel::setSourceModel(static_cast<QAbstractItemModel*>(model));
 }
 
+void FilterModel::sortByParameter()
+{
+    switch (m_sortParameter) {
+    case Name:
+        setSortRole(DocumentListModel::Roles::FileNameRole);
+        break;
+    case Type:
+        setSortRole(DocumentListModel::Roles::FileTypeRole);
+        break;
+    case Date:
+        setSortRole(DocumentListModel::Roles::FileDateRole);
+        break;
+    default:
+        break;
+    }
+
+    sort(0);
+}
+
+FilterModel::SortParameter FilterModel::sortParameter() const
+{
+    return m_sortParameter;
+}
+
+void FilterModel::setSortParameter(const SortParameter &sortParameter)
+{
+    m_sortParameter = sortParameter;
```

Comment:
We should emit signal only if value is actually changed and call sortByParameter() here, not from JS in QML.

Reply:
Modified